crates/matrix-sdk/src/event_cache/mod.rs

```diff
@@ -124,12 +120,9 @@ impl Debug for EventCache {
 
 impl EventCache {
     /// Create a new [`EventCache`] for the given client.
-    pub fn new(client: Client) -> Self {
-        let weak_client = Arc::downgrade(&client.inner);
-
+    pub fn new(client: &Client) -> Self {
```

Comment:
Now that we only use `client` to subscribe to room updates (I think?) I might be inclined to add a `subscribe` or similar method, and make `new` not need `client`. Then we could remove the `OnceCell` dance from the `Client` creation code. What do you think?

Reply:
Sounds great, also having the cache enabled by default is bad, it should be user-controlled and opt-in instead :+1: